rdmd.d

```diff
@@ -330,7 +330,8 @@ size_t indexOfProgram(string[] args)
     foreach(i, arg; args[1 .. $])
     {
         if (!arg.startsWith('-', '@') &&
-                !arg.endsWith(".obj", ".o", ".lib", ".a", ".def", ".map", ".res"))
+                !arg.endsWith(".obj", ".o", ".lib", ".a", ".def", ".map", ".res") &&
+                args[i /*- 1 + 1*/] != "--eval")
```

Comment:
This is already checked via `!arg.startsWith('-', '@')` isn't it?

Reply:
No, the new snippet allows --eval ```code```. Before it would think ```code``` was a file.

Reply:
oh I see.  `args[i] != arg`...very confusing I must say.  I see you put a comment to try to indicate this. 

Maybe create a new variable `previousArg` to make it less confusing?